src/ttauri/wfree_fifo.hpp

```diff
@@ -124,62 +128,48 @@ class wfree_fifo {
         // The division here should be eliminated by the equal multiplication inside the index operator.
         // auto &slot = _slots[index / slot_size];
 
-        // Wait until the slot.pointer is a nullptr.
-        // And acquire the buffer to start overwriting it.
-        // There are no other threads that will make this non-null afterwards.
-        while (slot.pointer.load(std::memory_order_acquire)) {
-            // If we get here, that would suck, but nothing to do about it.
-            [[unlikely]] contended();
-        }
-
-        // Overwrite the buffer with the new slot.
-        auto new_ptr = new (slot.buffer.data()) Message(std::forward<Args>(args)...);
+        Message* new_ptr;
```

Comment:
I try to avoid variable declarations if I can. I rather have the store at the end of the function duplicated in both the if and else and keep `new_ptr` local to the if and else scopes.

Reply:
Even in case of constexpr if?

Reply:
I don't see how a `constexpr if` changes this.

Reply:
I thought, that after optimizer is done doing its thing, the resulting assembly be the same:
https://godbolt.org/z/dcPPPn7vh